### PR TITLE
Replace fetch with wp.ajax.post

### LINKED ISSUE #125: Replace fetch with wp.ajax.post
### Is your enhancement related to a problem? Please describe.

Replace javascript fetches with `wp.ajax.post`, which is a helper javascript util to connect to admin ajax. 

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chriscct7**
Fixes #125

This PR was written by AI and has not been reviewed yet for accuracy

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/WordPress/plugin-check/issues/125?shareId=b23b38d0-5bf0-4b68-9504-e131b9d77586).

### CODE REVIEW

**Comment by chriscct7** [comment]
@ernilambar make sure you review this one thoroughly. We were testing an AI tool at Contributors Day that wrote the pr.

**Comment by ernilambar** [comment]
I tested this in my local setup and it is not working. Instead of returning data from AJAX, we get only Promise object.

**Comment by chriscct7** [comment]
Sounds good. I'll close this so we don't accidentally merge it
